plugins/csp-wms-overlays/src/WebMapService.hpp

```diff
@@ -0,0 +1,88 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////
+//                               This file is part of CosmoScout VR                               //
+//      and may be used under the terms of the MIT license. See the LICENSE file for details.     //
+//                        Copyright: (c) 2020 German Aerospace Center (DLR)                       //
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef CSP_WMS_OVERLAYS_WEB_MAP_SERVICE_HPP
+#define CSP_WMS_OVERLAYS_WEB_MAP_SERVICE_HPP
+
+#include "WebMapLayer.hpp"
+
+#include <VistaTools/tinyXML/tinyxml.h>
+
+#include <array>
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+namespace csp::wmsoverlays {
+
+/// Class for storing information on a Web Map Service.
+class WebMapService {
+ public:
+  /// Struct for storing general WMS settings.
+  struct Settings {
+    std::optional<int> mMaxWidth;
+    std::optional<int> mMaxHeight;
+  };
+
+  /// Create an object for a WMS accessible at the given URL.
+  /// The url string should be the base URL of the WMS without a query string.
+  /// cacheDir should be the path to a directory which will be used for
+  /// caching capability documents.
+  WebMapService(std::string url, std::string cacheDir);
+
+  /// Gets the base URL of the service
+  std::string getUrl() const;
+  /// Gets a brief description of the service.
+  std::string getTitle() const;
+
+  /// Gets the general settings of the service.
+  Settings getSettings() const;
+
+  /// Gets the root layer of the service.
+  WebMapLayer getRootLayer() const;
+  /// Gets a list of all layers of the service, for which maps can be requested.
+  std::vector<WebMapLayer> getLayers() const;
+  /// Gets the layer with the given name, if one exists.
+  /// Returns an empty optional otherwise.
+  std::optional<WebMapLayer> getLayer(std::string name) const;
```

Comment:
These should all return by `const&`, shouldn't they? And the last should take a `const&` string.

```suggestion
  std::string const& getUrl() const;
  /// Gets a brief description of the service.
  std::string const& getTitle() const;

  /// Gets the general settings of the service.
  Settings const& getSettings() const;

  /// Gets the root layer of the service.
  WebMapLayer const& getRootLayer() const;
  /// Gets a list of all layers of the service, for which maps can be requested.
  std::vector<WebMapLayer> const& getLayers() const;
  /// Gets the layer with the given name, if one exists.
  /// Returns an empty optional otherwise.
  std::optional<WebMapLayer> const& getLayer(std::string const& name) const;
```